tasks/system_probe.py

```diff
@@ -282,14 +283,33 @@ def kitchen_prepare(ctx):
 
 
 @task
-def kitchen_test(ctx):
+def kitchen_test(ctx, target=None):
     """
     Run tests (locally) using chef kitchen against an array of different platforms.
     * Make sure to run `inv -e system-probe.kitchen-prepare` using the agent-development VM;
     * Then we recommend to run `inv -e system-probe.kitchen-test` directly from your (macOS) machine;
     """
+
+    # Retrieve a list of all available vagrant images
+    images = {}
+    with open(os.path.join(KITCHEN_DIR, "platforms.json"), 'r') as f:
+        for platform, by_provider in json.load(f).items():
+            if "vagrant" in by_provider:
+                for image in by_provider["vagrant"]:
+                    images[image] = platform
+
+    if not (target in images):
+        print("please run inv -e system-probe.kitchen-test --target <IMAGE>, where <IMAGE> is one of the following:\n%s" % (list(images.keys())))
+        raise Exit(code=1)
+
     with ctx.cd(KITCHEN_DIR):
-        ctx.run("kitchen test", env={"KITCHEN_YAML": "kitchen-vagrant-system-probe.yml"})
+        ctx.run("inv kitchen.genconfig --platform {platform} --osversions {target} --provider vagrant --testfiles system-probe-test".format(
+            target=target,
+            platform=images[target]
+        ))
+        # In-place editting varies across different `sed` versions, so we write to a temporary file
+        ctx.run("sed '/provider:/d' kitchen.yml > kitchen.temp && mv kitchen.temp kitchen.yml")
```

Comment:
Rather than doing this, we should either set env var `KITCHEN_VAGRANT_PROVIDER=virtualbox` or remove this line: https://github.com/DataDog/datadog-agent/blob/c0a72474b98b580119a8ccc0dc30eb1a01f0cfc9/test/kitchen/drivers/vagrant-driver.yml#L4

Reply:
Awesome. I'll update it